core/test/Test/Pos/Core/Gen.hs

```diff
@@ -0,0 +1,940 @@
+genGenesisInitializer =
+    GenesisInitializer
+        <$> genTestnetBalanceOptions
+        <*> genFakeAvvmOptions
+        <*> genCoinPortion
+        <*> Gen.bool
+        <*> Gen.integral (Range.constant 0 10)
+
+genGenesisProtocolConstants :: Gen GenesisProtocolConstants
+genGenesisProtocolConstants =
+    GenesisProtocolConstants
+        <$> Gen.int (Range.constant 0 100)
+        <*> genProtocolMagic
+        <*> genVssMaxTTL
+        <*> genVssMinTTL
+
+genGenesisSpec :: Gen GenesisSpec
+genGenesisSpec = mkGenSpec >>=  either (error . toText) pure
+    where
+        mkGenSpec = mkGenesisSpec
+                      <$> genGenesisAvvmBalances
+                      <*> genSharedSeed
+                      <*> genGenesisDelegation
+                      <*> genBlockVersionData
+                      <*> genGenesisProtocolConstants
+                      <*> genGenesisInitializer
+
+genTestnetBalanceOptions :: Gen TestnetBalanceOptions
+genTestnetBalanceOptions =
+    TestnetBalanceOptions
+        <$> Gen.word Range.constantBounded
+        <*> Gen.word Range.constantBounded
+        <*> Gen.word64 Range.constantBounded
+        <*> Gen.double (Range.constant 0 10)
+        <*> Gen.bool
+----------------------------------------------------------------------------
+-- Pos.Core.ProtocolConstants Generators
+----------------------------------------------------------------------------
+
+genVssMaxTTL :: Gen VssMaxTTL
+genVssMaxTTL = VssMaxTTL <$> genWord32
+
+genVssMinTTL :: Gen VssMinTTL
+genVssMinTTL = VssMinTTL <$> genWord32
+
+----------------------------------------------------------------------------
+-- Pos.Core.Slotting Generators
+----------------------------------------------------------------------------
+
+genEpochIndex :: Gen EpochIndex
+genEpochIndex = EpochIndex <$> Gen.word64 Range.constantBounded
+
+genEpochOrSlot :: Gen EpochOrSlot
+genEpochOrSlot =
+    Gen.choice [ EpochOrSlot . Left <$> genEpochIndex
+               , EpochOrSlot . Right <$> genSlotId
+               ]
+
+genFlatSlotId :: Gen FlatSlotId
+genFlatSlotId = Gen.word64 Range.constantBounded
+
+genLocalSlotIndex :: Gen LocalSlotIndex
+genLocalSlotIndex = UnsafeLocalSlotIndex <$> Gen.word16 (Range.constant 0 21599)
+
+genSlotCount :: Gen SlotCount
+genSlotCount = SlotCount <$> Gen.word64 Range.constantBounded
+
+genSlotId :: Gen SlotId
+genSlotId = SlotId <$> genEpochIndex <*> genLocalSlotIndex
+
+genTimeDiff :: Gen TimeDiff
+genTimeDiff = TimeDiff <$> genMicrosecond
+
+genTimestamp :: Gen Timestamp
+genTimestamp = Timestamp <$> genMicrosecond
+
+----------------------------------------------------------------------------
+-- Pos.Core.Ssc Generators
+----------------------------------------------------------------------------
+
+genGenesisAvvmBalances :: Gen GenesisAvvmBalances
+genGenesisAvvmBalances = GenesisAvvmBalances <$> customHashMapGen genRedeemPublicKey genCoin
+
+data CommitmentOpening = CommitmentOpening
```

Comment:
This was intended to be a sort of "helper datatype" just so that I didn't have to deal with the typical return type of [genCommitmentAndOpening](https://github.com/input-output-hk/cardano-sl/blob/42b22ea6f19ef784a4159447ea77c85e035df3f0/ssc/Pos/Ssc/Base.hs#L82).

Reply:
Cool, makes sense.

Reply:
This isn't a big deal for me to change, though. I could modify this and work with the `(Commitment, Opening)` type that is returned instead.